couchdb/server/compiler.py

```diff
@@ -17,6 +17,24 @@
 log = logging.getLogger(__name__)
 
 
+class EggExports(dict):
+    """Sentinel for egg export statements."""
+
+
+def maybe_b64egg(b64str):
+    """Checks if passed string is base64 encoded egg file"""
+    # Quick and dirty check for base64 encoded zipfile.
+    # Saves time and IO operations in most cases.
+    return isinstance(b64str, util.strbase) and b64str.startswith('UEsDBBQAAAAIA')
```

Comment:
The `UEsDBBQAAAAIA` is about egg file magic number. The point was to not try to b64decode that string if it even doesn't looks like a base64 encoded egg file. Because encoded eggs are quite heavy, you'll do double decoding to get it loaded with your solution.